pkg/services/heartbeat.go

```diff
@@ -0,0 +1,59 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/timeutil"
+)
+
+// Heartbeat is a usage of Engine for application specific heartbeats,
+// used in the core node and for loops. It accepts a named logger,
+// a beat, a setup func to initialize resources used on each beat,
+// a beat function to define the behavior on each beat, and a close func
+// for resource teardown
+type Heartbeat struct {
+	Service
+	eng *Engine
+
+	beat time.Duration
+	lggr logger.Logger
```

Comment:
Shall we pass beholder client as a dependency here as well. Its same type of concern as logger.